src/style/style.js

```diff
@@ -568,7 +568,10 @@ class Style extends Evented {
 
         layer.setEventedParent(this, {layer: {id: id}});
 
-        const index = before ? this._order.indexOf(before) : this._order.length;
+        const beforeIndex = before ? this._order.indexOf(before) : 0;
```

Comment:
@mollymerp is `0` the top or bottom of the stack? Hitting `map.style._order` in my console, the first layer in the array is visually under all other layers on the map.

Reply:
I suppose this is misleading – `beforeIndex` isn't actually used unless `before` exists so the `0` value never gets assigned to `index`. I made it 0 instead of `null` to appease flow but I could make this clearer.